components/brave_ads/core/internal/ads/serving/eligible_ads/pipelines/notification_ads/eligible_notification_ads_v2.cc

```diff
@@ -134,10 +135,15 @@ CreativeNotificationAdList EligibleNotificationAdsV2::FilterCreativeAds(
     return {};
   }
 
+  CreativeNotificationAdList eligible_creative_ads = creative_ads;
```

Comment:
This is an extra copy. Maybe it is better to do:

```
  CreativeInlineContentAdList eligible_creative_ads = ApplyExclusionRules(
      creative_ads, last_served_ad_, &exclusion_rules);

```

Reply:
Great catch. Fixed.